### DISCUSSION TITLE
Foster Docs

### DISCUSSION THREAD

**Discussion by sebastianwhiffen** [FosterFramework/Foster#97] [Ideas]
Really enjoying this framework as a c# developer, but it could seriously use some docs!

Two things that I would be appreciative of as a user would be:
1. a wiki containing some of the usual stuff like 'getting started' and a description of anything exposed to the public API
2. XML docs so when I hover over a Foster method or class I can tell what its intended use is

I'd be more than happy to commit these changes myself if there was some interest in it : )

The XML comments are straight forward enough, but perhaps a group decision should be made on how we present the official docs

**Comment by sebastianwhiffen** [comment]
My bad on the XML comment part


took a look at the source and it seems like there's already comments, not sure why my editor isn't showing them?

![image](https://github.com/user-attachments/assets/e1aed146-b0a0-4feb-9deb-2df4f0245809)

**Reply by NoelFB** [reply]
Interesting... I actually wonder if they're somehow not being included in the NuGet package? I'll have to take a look!

**Reply by sebastianwhiffen** [reply]
I was able to fix it by adding:
GenerateDocumentationFile = true 
to the csproj file, clearing my nuget cache and then re-packing the file to somewhere on my local. [PR](https://github.com/FosterFramework/Foster/pull/98)

![image](https://github.com/user-attachments/assets/8db26202-64e3-4861-b5eb-3a0f71d182c8)



this XML flag would also be needed for [ttps://dotnet.github.io/docfx/](https://dotnet.github.io/docfx/) if you were to decide you wanted to use that

**Comment by NoelFB** [comment]
Yeah I would like to make a simple static site that uses the XML docs (which could also be improved). There's one specifically for dotnet that might be a good choice: https://dotnet.github.io/docfx/

**Reply by sebastianwhiffen** [reply]
maybe set out a 'standard' you'd like public methods to have for their comments, like a minimum requirements so they don't get sloppy if someone comes to write new / re-write old ones

**Comment by landon-codes** [comment]
A wiki would be super easy for community members to manage documentation so the owner can focus on making their tools. Though GitHub recommends monitoring a repo wiki so maybe it's not the best idea.

Regardless I think documentation would be amazing, but I understand why there isn't any since this project is so early. It's better to focus on features at the moment. 

I do know that I didn't feel like all the comments were very good while I was looking up the definitions of functions or types when going through the examples. Often, they would just restate the name of the function.

**Comment by NoelFB** [comment]
Yeah, I agree a lot of the comments need work! There has been some community-created docs but nothing that has been moved into the main repos yet.

**Comment by DevTwilight** [comment]
i suggest GitHub pages better than wiki
